### ISSUE TITLE
Tensorflow Prediction Artifacts

### ISSUE THREAD

**Issue by tibuch** [tensorflow/tensorflow#21637]
### System Information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Mint 18
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**: N/A
- **TensorFlow installed from (source or binary)**: Binary
- **TensorFlow version (use command below)**: v1.10.0-0-g656e7a2b34 1.10.0
- **Python version**: 3.6
- **Bazel version (if compiling from source)**: N/A
- **GCC/Compiler version (if compiling from source)**: N/A
- **CUDA/cuDNN version**: 9.0.176/7.1.2
- **GPU model and memory**: Tesla P40 24GB
- **Exact command to reproduce**: python3 example.py

### Describe the problem
Hello,
I am using Tensorflow on 3D image data and I observe some strange prediction artifacts once my input exceeds a certain size.
I created a minimal working example (below) which produces the artifacts. In the example all weights of the loaded u-net are initialized to 0.5 and then a 3D volume (Z: 180, Y: 196, X: 240) with a gradient in Z direction is given as prediction input. The returned output contains the gradient up to Z around 140 and then the output values change to smaller values. 
![observation](https://user-images.githubusercontent.com/5450869/44163361-7661fc00-a0c3-11e8-8c67-a3524370b632.png)

If a slightly smaller 3D volume (Z: 176, Y: 196, X: 240) is given, the output contains the full gradient as expected. 

Unfortunately I don't know if this is a TensorFlow, CUDA, cuDNN or some other sort of bug. I am grateful for any explanation or any hints on what it could be or how to fix this issue (without cropping/tiling the input).

### Source code / logs

The minimum example:
```
import tensorflow as tf
from tensorflow.keras.models import model_from_json
import numpy as np
from tifffile import imsave

# load u-net architecture, kernel-size = 3, depth = 2
json_file = open('model.json', 'r')
loaded_model_json = json_file.read()
json_file.close()
model = model_from_json(loaded_model_json)

# set all weights to 0.5
w = model.get_weights()
init = []
for l in w:
    init.append(l*0 + 0.5)

model.set_weights(init)

tmp = np.zeros((180, 196, 240), np.float32)
for i in range(tmp.shape[0]):
    tmp[i] = i

tmp = tmp[:, :, :, np.newaxis]

prediction_fail = np.moveaxis(model.predict(np.moveaxis(tmp[np.newaxis], 4, -1))[0], 3, 0)
tmp = tmp[:176]
prediction_succ = np.moveaxis(model.predict(np.moveaxis(tmp[np.newaxis], 4, -1))[0], 3, 0)

imsave('input.tif', tmp)
imsave('prediction_fail.tif', prediction_fail[0])
imsave('prediction_success.tif', prediction_succ[0])
```

The model.json with the u-net architecture:
```
{"class_name": "Model", "config": {"name": "model_3", "layers": [{"name": "input", "class_name": "InputLayer", "config": {"batch_input_shape": [null, null, null, null, 1], "dtype": "float32", "sparse": false, "name": "input"}, "inbound_nodes": []}, {"name": "down_level_0_no_0", "class_name": "Conv3D", "config": {"name": "down_level_0_no_0", "trainable": true, "filters": 32, "kernel_size": [3, 3, 3], "strides": [1, 1, 1], "padding": "same", "data_format": "channels_last", "dilation_rate": [1, 1, 1], "activation": "relu", "use_bias": true, "kernel_initializer": {"class_name": "VarianceScaling", "config": {"scale": 1.0, "mode": "fan_avg", "distribution": "uniform", "seed": null}}, "bias_initializer": {"class_name": "Zeros", "config": {}}, "kernel_regularizer": null, "bias_regularizer": null, "activity_regularizer": null, "kernel_constraint": null, "bias_constraint": null}, "inbound_nodes": [[["input", 0, 0, {}]]]}, {"name": "down_level_0_no_1", "class_name": "Conv3D", "config": {"name": "down_level_0_no_1", "trainable": true, "filters": 32, "kernel_size": [3, 3, 3], "strides": [1, 1, 1], "padding": "same", "data_format": "channels_last", "dilation_rate": [1, 1, 1], "activation": "relu", "use_bias": true, "kernel_initializer": {"class_name": "VarianceScaling", "config": {"scale": 1.0, "mode": "fan_avg", "distribution": "uniform", "seed": null}}, "bias_initializer": {"class_name": "Zeros", "config": {}}, "kernel_regularizer": null, "bias_regularizer": null, "activity_regularizer": null, "kernel_constraint": null, "bias_constraint": null}, "inbound_nodes": [[["down_level_0_no_0", 0, 0, {}]]]}, {"name": "max_0", "class_name": "MaxPooling3D", "config": {"name": "max_0", "trainable": true, "pool_size": [2, 2, 2], "padding": "valid", "strides": [2, 2, 2], "data_format": "channels_last"}, "inbound_nodes": [[["down_level_0_no_1", 0, 0, {}]]]}, {"name": "down_level_1_no_0", "class_name": "Conv3D", "config": {"name": "down_level_1_no_0", "trainable": true, "filters": 64, "kernel_size": [3, 3, 3], "strides": [1, 1, 1], "padding": "same", "data_format": "channels_last", "dilation_rate": [1, 1, 1], "activation": "relu", "use_bias": true, "kernel_initializer": {"class_name": "VarianceScaling", "config": {"scale": 1.0, "mode": "fan_avg", "distribution": "uniform", "seed": null}}, "bias_initializer": {"class_name": "Zeros", "config": {}}, "kernel_regularizer": null, "bias_regularizer": null, "activity_regularizer": null, "kernel_constraint": null, "bias_constraint": null}, "inbound_nodes": [[["max_0", 0, 0, {}]]]}, {"name": "down_level_1_no_1", "class_name": "Conv3D", "config": {"name": "down_level_1_no_1", "trainable": true, "filters": 64, "kernel_size": [3, 3, 3], "strides": [1, 1, 1], "padding": "same", "data_format": "channels_last", "dilation_rate": [1, 1, 1], "activation": "relu", "use_bias": true, "kernel_initializer": {"class_name": "VarianceScaling", "config": {"scale": 1.0, "mode": "fan_avg", "distribution": "uniform", "seed": null}}, "bias_initializer": {"class_name": "Zeros", "config": {}}, "kernel_regularizer": null, "bias_regularizer": null, "activity_regularizer": null, "kernel_constraint": null, "bias_constraint": null}, "inbound_nodes": [[["down_level_1_no_0", 0, 0, {}]]]}, {"name": "max_1", "class_name": "MaxPooling3D", "config": {"name": "max_1", "trainable": true, "pool_size": [2, 2, 2], "padding": "valid", "strides": [2, 2, 2], "data_format": "channels_last"}, "inbound_nodes": [[["down_level_1_no_1", 0, 0, {}]]]}, {"name": "middle_0", "class_name": "Conv3D", "config": {"name": "middle_0", "trainable": true, "filters": 128, "kernel_size": [3, 3, 3], "strides": [1, 1, 1], "padding": "same", "data_format": "channels_last", "dilation_rate": [1, 1, 1], "activation": "relu", "use_bias": true, "kernel_initializer": {"class_name": "VarianceScaling", "config": {"scale": 1.0, "mode": "fan_avg", "distribution": "uniform", "seed": null}}, "bias_initializer": {"class_name": "Zeros", "config": {}}, "kernel_regularizer": null, "bias_regularizer": null, "activity_regularizer": null, "kernel_constraint": null, "bias_constraint": null}, "inbound_nodes": [[["max_1", 0, 0, {}]]]}, {"name": "middle_2", "class_name": "Conv3D", "config": {"name": "middle_2", "trainable": true, "filters": 64, "kernel_size": [3, 3, 3], "strides": [1, 1, 1], "padding": "same", "data_format": "channels_last", "dilation_rate": [1, 1, 1], "activation": "relu", "use_bias": true, "kernel_initializer": {"class_name": "VarianceScaling", "config": {"scale": 1.0, "mode": "fan_avg", "distribution": "uniform", "seed": null}}, "bias_initializer": {"class_name": "Zeros", "config": {}}, "kernel_regularizer": null, "bias_regularizer": null, "activity_regularizer": null, "kernel_constraint": null, "bias_constraint": null}, "inbound_nodes": [[["middle_0", 0, 0, {}]]]}, {"name": "up_sampling3d_5", "class_name": "UpSampling3D", "config": {"name": "up_sampling3d_5", "trainable": true, "size": [2, 2, 2], "data_format": "channels_last"}, "inbound_nodes": [[["middle_2", 0, 0, {}]]]}, {"name": "concatenate_5", "class_name": "Concatenate", "config": {"name": "concatenate_5", "trainable": true, "axis": -1}, "inbound_nodes": [[["up_sampling3d_5", 0, 0, {}], ["down_level_1_no_1", 0, 0, {}]]]}, {"name": "up_level_1_no_0", "class_name": "Conv3D", "config": {"name": "up_level_1_no_0", "trainable": true, "filters": 64, "kernel_size": [3, 3, 3], "strides": [1, 1, 1], "padding": "same", "data_format": "channels_last", "dilation_rate": [1, 1, 1], "activation": "relu", "use_bias": true, "kernel_initializer": {"class_name": "VarianceScaling", "config": {"scale": 1.0, "mode": "fan_avg", "distribution": "uniform", "seed": null}}, "bias_initializer": {"class_name": "Zeros", "config": {}}, "kernel_regularizer": null, "bias_regularizer": null, "activity_regularizer": null, "kernel_constraint": null, "bias_constraint": null}, "inbound_nodes": [[["concatenate_5", 0, 0, {}]]]}, {"name": "up_level_1_no_2", "class_name": "Conv3D", "config": {"name": "up_level_1_no_2", "trainable": true, "filters": 32, "kernel_size": [3, 3, 3], "strides": [1, 1, 1], "padding": "same", "data_format": "channels_last", "dilation_rate": [1, 1, 1], "activation": "relu", "use_bias": true, "kernel_initializer": {"class_name": "VarianceScaling", "config": {"scale": 1.0, "mode": "fan_avg", "distribution": "uniform", "seed": null}}, "bias_initializer": {"class_name": "Zeros", "config": {}}, "kernel_regularizer": null, "bias_regularizer": null, "activity_regularizer": null, "kernel_constraint": null, "bias_constraint": null}, "inbound_nodes": [[["up_level_1_no_0", 0, 0, {}]]]}, {"name": "up_sampling3d_6", "class_name": "UpSampling3D", "config": {"name": "up_sampling3d_6", "trainable": true, "size": [2, 2, 2], "data_format": "channels_last"}, "inbound_nodes": [[["up_level_1_no_2", 0, 0, {}]]]}, {"name": "concatenate_6", "class_name": "Concatenate", "config": {"name": "concatenate_6", "trainable": true, "axis": -1}, "inbound_nodes": [[["up_sampling3d_6", 0, 0, {}], ["down_level_0_no_1", 0, 0, {}]]]}, {"name": "up_level_0_no_0", "class_name": "Conv3D", "config": {"name": "up_level_0_no_0", "trainable": true, "filters": 32, "kernel_size": [3, 3, 3], "strides": [1, 1, 1], "padding": "same", "data_format": "channels_last", "dilation_rate": [1, 1, 1], "activation": "relu", "use_bias": true, "kernel_initializer": {"class_name": "VarianceScaling", "config": {"scale": 1.0, "mode": "fan_avg", "distribution": "uniform", "seed": null}}, "bias_initializer": {"class_name": "Zeros", "config": {}}, "kernel_regularizer": null, "bias_regularizer": null, "activity_regularizer": null, "kernel_constraint": null, "bias_constraint": null}, "inbound_nodes": [[["concatenate_6", 0, 0, {}]]]}, {"name": "up_level_0_no_2", "class_name": "Conv3D", "config": {"name": "up_level_0_no_2", "trainable": true, "filters": 32, "kernel_size": [3, 3, 3], "strides": [1, 1, 1], "padding": "same", "data_format": "channels_last", "dilation_rate": [1, 1, 1], "activation": "relu", "use_bias": true, "kernel_initializer": {"class_name": "VarianceScaling", "config": {"scale": 1.0, "mode": "fan_avg", "distribution": "uniform", "seed": null}}, "bias_initializer": {"class_name": "Zeros", "config": {}}, "kernel_regularizer": null, "bias_regularizer": null, "activity_regularizer": null, "kernel_constraint": null, "bias_constraint": null}, "inbound_nodes": [[["up_level_0_no_0", 0, 0, {}]]]}, {"name": "conv3d_3", "class_name": "Conv3D", "config": {"name": "conv3d_3", "trainable": true, "filters": 1, "kernel_size": [1, 1, 1], "strides": [1, 1, 1], "padding": "valid", "data_format": "channels_last", "dilation_rate": [1, 1, 1], "activation": "linear", "use_bias": true, "kernel_initializer": {"class_name": "VarianceScaling", "config": {"scale": 1.0, "mode": "fan_avg", "distribution": "uniform", "seed": null}}, "bias_initializer": {"class_name": "Zeros", "config": {}}, "kernel_regularizer": null, "bias_regularizer": null, "activity_regularizer": null, "kernel_constraint": null, "bias_constraint": null}, "inbound_nodes": [[["up_level_0_no_2", 0, 0, {}]]]}, {"name": "add_19", "class_name": "Add", "config": {"name": "add_19", "trainable": true}, "inbound_nodes": [[["conv3d_3", 0, 0, {}], ["input", 0, 0, {}]]]}, {"name": "activation_52", "class_name": "Activation", "config": {"name": "activation_52", "trainable": true, "activation": "linear"}, "inbound_nodes": [[["add_19", 0, 0, {}]]]}], "input_layers": [["input", 0, 0]], "output_layers": [["activation_52", 0, 0]]}, "keras_version": "2.2.0", "backend": "tensorflow"}
```

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
Bazel version
GPU model and memory
Exact command to reproduce
Mobile device

**Comment by tomasvicar** [comment]
Hello

I am observing same behavior (of example.py) on a totally different system:

OS Platform: Windows 10
CUDA/cuDNN version: 9.0.176/7.1.4
TensorFlow installed from (source or binary): source
TensorFlow version: 1.9.0
Python version: 3.6
GPU model and memory: nvidia geforce gtx 1070, 8 GB

**Comment by poxvoculi** [comment]
@fchollet Can you take a look at this?  Is it an incompatibility between the input size and the model definition, or does it look like a bug somewhere?

**Comment by iteachmachines** [comment]
Hello even i am facing the same issue on windows 10,can anyone help?

**Comment by psteinb** [comment]
for the record, I tested this code in a tf 1.8 and 1.10 container obtained from [dockerhub](https://hub.docker.com/r/tensorflow/tensorflow/) with Nvidia driver 384.145 and the artifact cannot be seen!

I then did the same in a tf installation, produced by following the suggested method in the [online tensorflow documentation](https://www.tensorflow.org/install/install_linux#InstallingVirtualenv). Interestingly the artifact did appear!

**Comment by psteinb** [comment]
same setup with virtualenv and tf1.10, a TitanXp and nvidia driver 390.42 ... no artifact (also using the containers mentioned above). could it be the wheels contain problematic ptx for the P40?

**Comment by lucncdm** [comment]
Hello, i also encountered this issue.
I isolated the layer in my network that produced this artefact: it comes from the operation Conv3D.
It doesn't occur for every input size, output channel, or kernel size.

Here is the code i used to reproduce the behaviour:
```
from keras.models import Model
from keras.layers import Input, Conv3D
import numpy as np
import matplotlib.pyplot as plt

def conv_model_kernel_1(in_channel, out_channel):
    inputs = Input((None, None, None, in_channel))
    output = Conv3D(out_channel, kernel_size=1, padding='same', kernel_initializer='ones', bias_initializer='zeros')(inputs)
    return Model(inputs, output)

channels = 32
volume = np.ones((202, 236, 185, channels))
net = conv_model_kernel_1(channels, channels)
conv_keras = net.predict(np.expand_dims(volume, 0))

plt.figure()
plt.imshow(conv_keras[0, :, :, 0, 0])
plt.show()
```

The network as defined previously should take every input channel and sum them in each output channel. Each voxel of each volume contained in every output channel should have the same value : 32.

Here the slice 0 of the output predicted volume contained in channel 0:
![conv3dartifact](https://user-images.githubusercontent.com/24569806/44464042-eb7c8680-a619-11e8-969a-809973f84650.png)

Yellow pixels are 32s, purple ones are 0s.

I tested this code on different system configurations and everytime it produced the artifact.

I'd be thankful if you could take a look.
Thanks

**Comment by tibuch** [comment]
Hi @lucncdm 
That is an even better minimal example!
Is your tf installation in an environment or is it a docker? It looks like the artifact is __not__ appearing in the docker.

**Comment by lucncdm** [comment]
Hi @tibuch 
I compiled tensorflow from source using bazel. I didn't try using a docker yet.
Here is my system configuration, just in case:
CentOs
tensorflow: 1.7
keras: 2.0.9
cuda: 9.1
cudnn: 7.1
nvidia driver: 390.48

**Comment by tensorflowbutler** [comment]
Nagging Assignees @fchollet, @poxvoculi: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @fchollet, @poxvoculi: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @fchollet, @poxvoculi: It has been 74 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by maweigert** [comment]
Hi, 

I've seen the same prediction artifacts with CUDA 9.0, tf 1.12, cudnn 7.4.2 and even for more innocuous input shapes (e.g. `96, 480, 480`). **The solution was to install all the CUDA 9.0 patches** that are listed at the [installation page](https://developer.nvidia.com/cuda-90-download-archive?target_os=Linux&target_arch=x86_64&target_distro=Ubuntu&target_version=1704&target_type=runfilelocal). Afterwards, all was back to normal.

**Comment by mohantym** [comment]
Hi @tibuch !
We are checking to see whether you still need help in this issue . Have you tried latest versions TF 2.6/2.7 yet? . Thanks!

**Comment by tibuch** [comment]
I tried out the small example of @lucncdm with CUDA 10.1.243, CUDNN 7.6.5 and TensorFlow 2.3.0 on a NVIDIA GeForce GTX 1080 Ti with NVIDIA driver 470.57.02 and the error did __not__ show up.

I also tried both examples with TF 2.7 CPU and the error did not show up.

**Comment by mohantym** [comment]
Ok! @tibuch ,Thanks for confirming the same! Closing this issue as it seems to be resolved .
